### PR TITLE
Improve rosetta docs

### PR DESCRIPTION
**Pull request by qwadratic**

### CODE REVIEW

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/build-from-sources.mdx:18]:
```suggestion
In [Running with Docker](run-with-docker), you learned that the Docker container runs the mina daemon, mina-archive, and Rosetta API. 
```

Comment (barriebyron) [reply docs/exchange-operators/rosetta/build-from-sources.mdx:18]: thanks for this PR, here's some resources you can rely on https://github.com/o1-labs/docs2/wiki/Docs-Style-Guide 

- instead of "we" third person, docs style is to use second person "you" pronoun 
- instead of "previous article", use descriptive link text
- we love the serial comma!

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/build-from-sources.mdx:20]:
```suggestion
If you already have mina-archive up and running, you can also build and run the Rosetta API natively and wire it to the existing mina daemon and mina-archive.
```

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/build-from-sources.mdx:24]:
```suggestion
To run mina daemon and mina-archive natively, follow the [Archive Node page](node-operators/archive-node) instructions.
```

Comment (barriebyron) [reply docs/exchange-operators/rosetta/build-from-sources.mdx:24]: suggest using the same terminology for all references to "archive"

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/build-from-sources.mdx:28]:
```suggestion
The easiest way to build Rosetta natively is to use the Nix development environment.
```

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/build-from-sources.mdx:33]:
```suggestion
1. Clone the official [mina repository](https://github.com/MinaProtocol/mina.git) and switch to the `compatible` branch.

2. If you don't already have Nix on your machine, install it following the steps in the [nix/README.md](https://github.com/MinaProtocol/mina/blob/develop/nix/README.md).

```

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/build-from-sources.mdx:52]:
```suggestion
6. After a successful build, the Rosetta app is available here: `_build/default/src/app/rosetta/rosetta.exe` 
```

File: docs/exchange-operators/rosetta/index.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/index.mdx:20]:
```suggestion
Learn how to use Mina’s implementation of the Rosetta API.
```

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/build-from-sources.mdx:22]:
```suggestion
:::tip
```

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/build-from-sources.mdx:43]: this Markdown syntax seems to work, if you want to follow the established style, apply the syntax described in https://github.com/o1-labs/docs2/wiki/Markdown-Syntax#numbered-steps

File: docs/exchange-operators/rosetta/run-with-docker.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/run-with-docker.mdx:21]:
```suggestion
If you want to run Mina Rosetta on the Apple silicon, you can use the steps in [Building and running Rosetta from source code](build-from-sources.
```

File: docs/exchange-operators/rosetta/run-with-docker.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/run-with-docker.mdx:20]:
```suggestion
There is a known issue that you can't use the docker image on the Apple silicon chip.
```

File: docs/exchange-operators/rosetta/run-with-docker.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/run-with-docker.mdx:27]:
```suggestion
2. Check the latest release for Mainnet on the official [Mina GitHub releases](https://github.com/MinaProtocol/mina/releases) page. 
```

Comment (barriebyron) [reply docs/exchange-operators/rosetta/run-with-docker.mdx:27]: hi @qwadratic  did you know we have a word list? (we always misspell GitHub) https://github.com/o1-labs/docs2/wiki/Word-list

File: docs/exchange-operators/rosetta/run-with-docker.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/run-with-docker.mdx:29]:
```suggestion
3. Go to [minaprotocol/mina-rosetta](https://hub.docker.com/r/minaprotocol/mina-rosetta) ` page on the Docker Hub repository. Pull the image with the commit hash matching to the latest Mainnet release.
```

File: docs/exchange-operators/rosetta/run-with-docker.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/run-with-docker.mdx:37]:
```suggestion
4. Use this image:

:::note

If you want to build your own docker image, you can find more details in [Mina’s Rosetta repository](https://github.com/MinaProtocol/mina/blob/develop/src/app/rosetta/README.md). However, for most users, it’s not necessary to build your own image in order to interact with the API.

:::
```

File: docs/exchange-operators/rosetta/samples/index.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/index.mdx:28]:
```suggestion
Start with the required imports and define constants:
```

File: docs/exchange-operators/rosetta/samples/index.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/index.mdx:53]:
```suggestion
To work with JSON requests and responses, you might want to print highlighted and indented JSON objects. You can define a simple function with the help of the `jq` utility:
```

File: docs/exchange-operators/rosetta/samples/requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/requests.mdx:12]:
```suggestion
The Rosetta API specification defines high-level description of how requests and response objects should look like.
Exact json object layouts are different from blockchain to blockchain.

In this article, you learn about Mina-specific Rosetta JSON objects and how to implement helper functions to construct and parse the JSON objects.
```

File: docs/exchange-operators/rosetta/samples/requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/requests.mdx:18]:
```suggestion
There is the `network_identifier` object, that should be passed as a parameter to all endpoints (except `/network/list` which returns possible identifiers).

First, implement the generic `request` function to perform Rosetta API calls injecting that object under the hood:
```

File: docs/exchange-operators/rosetta/samples/requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/requests.mdx:45]:
```suggestion
To implement helpers to construct the objects:
```

File: docs/exchange-operators/rosetta/samples/requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/requests.mdx:80]:
```suggestion
<details><summary>A sample JSON object that represents an operation</summary>
```

File: docs/exchange-operators/rosetta/samples/requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/requests.mdx:119]:
```suggestion
In Mina, each operation represents an [Account Update](/zkapps/o1js/interact-with-mina#account-updates). Therefore, a MINA token transfer transaction is represented by three Account Updates:
- for decreasing fee payer account balance (the fee payer's account is the same as the sender's)
```

File: docs/exchange-operators/rosetta/samples/requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/requests.mdx:123]:
```suggestion
<details><summary>A sample object that represents a transfer transaction</summary>
```

File: docs/exchange-operators/rosetta/samples/requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/requests.mdx:202]:
```suggestion
According to the Rosetta specification, the array of `operation` objects must be passed as a parameter in the `/construction/preprocess` and `/construction/payloads` endpoints.

To implement a helper function to construct the operations array for a MINA token transfer:

```

File: docs/exchange-operators/rosetta/samples/requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/requests.mdx:249]:
```suggestion
Now that you have all helpers in place, you can implement all of the Rosetta client methods according to the specification:
```

File: docs/exchange-operators/rosetta/samples/scan-blocks.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/scan-blocks.mdx:2]:
```suggestion
title: Scanning Blocks
```

File: docs/exchange-operators/rosetta/samples/scan-blocks.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/scan-blocks.mdx:6]:
```suggestion
description: Learn how to scan blocks and inspect transactions
---

To implement a simple function to wait for a given block:
```

File: docs/exchange-operators/rosetta/samples/send-transactions.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/send-transactions.mdx:3]:
```suggestion
description: Learn how to send transactions using a key pair
```

File: docs/exchange-operators/rosetta/samples/send-transactions.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/send-transactions.mdx:10]:
```suggestion
The steps to send a payment in MINA tokens:
```

File: docs/exchange-operators/rosetta/samples/send-transactions.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/send-transactions.mdx:11]: I won't mark the rest instances, but be sure to use two words for "key pair' 
https://github.com/o1-labs/docs2/wiki/Word-list#key-pair

File: docs/exchange-operators/rosetta/samples/send-transactions.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/send-transactions.mdx:22]:
```suggestion
For ease of readability, this sample implementation skips the sanity checks (steps 4 and 7) and combines steps 2 and 3 in a single `tx_payloads` function call.
```

File: docs/exchange-operators/rosetta/samples/track-deposits.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/track-deposits.mdx:9]:
```suggestion
After everything is ready, you can implement deposit tracking.

For that, iterate through all transactions in each block to search for `payment_receiver_inc` operation with an address matching the address in which you are interested:

```

File: docs/exchange-operators/rosetta/samples/using-signer.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/using-signer.mdx:10]:
```suggestion
According to the Rosetta specification, you use an offline tool to sign transactions and use Rosetta to submit the signed transactions.
The Mina Rosetta implementation comes with an offline signer tool. To generate a key pair, sign a transaction and verify a signature with it.

```

File: docs/exchange-operators/rosetta/samples/using-signer.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/using-signer.mdx:15]:
```suggestion
To simplify examples, it is referred to as the `signer` alias.
```

File: docs/exchange-operators/rosetta/samples/using-signer.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/using-signer.mdx:37]:
```suggestion
Replace `<tx>` with the unsigned transaction string, which you can construct using `construction/payloads` endpoint.
```

File: docs/exchange-operators/rosetta/samples/using-signer.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/using-signer.mdx:38]:
```suggestion
See [Sending Transactions](send-transactions) page to learn more.
```

File: docs/exchange-operators/rosetta/samples/using-signer.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/samples/using-signer.mdx:43]:
```suggestion
To implement a helper function to sign a transaction:
```

File: docs/exchange-operators/rosetta/send-requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/send-requests.mdx:3]:
```suggestion
description: Learn about Mina-specific request/response models
```

File: docs/exchange-operators/rosetta/send-requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/send-requests.mdx:16]:
```suggestion
After the Rosetta app is running, you can send requests to it.
To run a `/network/list` request:

    curl -L -X POST 'http://localhost:3087/network/list' \
        -H 'Content-Type: application/json' \
        -H 'Accept: application/json' -d '{}'

In case of succesfull setup, you should see a response similar to the one below:

```

File: docs/exchange-operators/rosetta/send-requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/send-requests.mdx:22]:
```suggestion
You must pass the `network_identifier` object as a parameter to all other endpoints. In Mina Rosetta implementation, it exists only for the network you run Rosetta for, so this array always contains one object.
```

Comment (barriebyron) [reply docs/exchange-operators/rosetta/send-requests.mdx:22]: present tense 
In general, use [present tense](https://developers.google.com/style/tense) rather than future tense; in particular, try to avoid using will where possible.

File: docs/exchange-operators/rosetta/send-requests.mdx
Comment (barriebyron) [inline docs/exchange-operators/rosetta/send-requests.mdx:24]:
```suggestion
Learn how to use Rosetta API for different use cases. See in the [Code Samples](samples) section. 
```

File: examples/exchange-operators/rosetta_samples.py
Comment (barriebyron) [reply examples/exchange-operators/rosetta_samples.py:11]: jq and Nix come up as prerequisites, suggest defining all prerequisites in a single section in the first doc: RAM, Nix, jq and so on

File: examples/exchange-operators/rosetta_samples.py
Comment (barriebyron) [inline examples/exchange-operators/rosetta_samples.py:98]:
```suggestion
The following helpers are for generating various objects in a Rosetta format, namely:
    - Identifier objects for block, tx, and account
    - array of operation objects to perform a native MINA token transfer
    - Public key object
```

File: examples/exchange-operators/rosetta_samples.py
Comment (barriebyron) [inline examples/exchange-operators/rosetta_samples.py:167]:
```suggestion
The following functions implement all requests 
available in the Mina Rosetta API
```

File: examples/exchange-operators/rosetta_samples.py
Comment (barriebyron) [inline examples/exchange-operators/rosetta_samples.py:253]:
```suggestion
The following examples are for typical operations that Rosetta API users might want to do:
```

**Review by barriebyron — APPROVED** [review]
thanks Ivan for these awesome docs!
To not hold you up, I've given conditional approval, see my comments and accept the suggestions.

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (qwadratic) [reply docs/exchange-operators/rosetta/build-from-sources.mdx:43]: @barriebyron it doesn't work when numbered steps are interrupted with code snippets

but I willl change it in another places. Thank you

File: docs/exchange-operators/rosetta/samples/send-transactions.mdx
Comment (qwadratic) [reply docs/exchange-operators/rosetta/samples/send-transactions.mdx]: @barriebyron done

**Comment by qwadratic** [comment]
@barriebyron I will need review from technical guys , so don't merge it yet please

File: docs/exchange-operators/rosetta/index.mdx
Comment (Sventimir) [inline docs/exchange-operators/rosetta/index.mdx:28]: I would say that using Rosetta is preferable in any case that Rosetta can cover. It's API is more appropriate for applications and improtantly, far more stable than GraphQL. The latter should only be used if there's no other choice. Sadly Rosetta does not cover mina-specific features like ZK apps.

File: docs/exchange-operators/rosetta/run-with-docker.mdx
Comment (Sventimir) [inline docs/exchange-operators/rosetta/run-with-docker.mdx:21]: This is being dealt with by @joaosreis as we speak. Please consult him for an update on this issue.

File: docs/exchange-operators/rosetta/samples/index.mdx
Comment (Sventimir) [inline docs/exchange-operators/rosetta/samples/index.mdx:20]: I don't think this is the case. I have just checked a 22.04 LTS Ubuntu installation and it does not include `jq` by default. Arch Linux that I use does not ship with it either. It needs to be installed manually.

File: docs/exchange-operators/rosetta/samples/index.mdx
Comment (Sventimir) [inline docs/exchange-operators/rosetta/samples/index.mdx:53]: I don't think `jq` is necessary for this. Python itself contains functionality to pretty-print JSON:
```
import json

json.dumps(j, indent=2)
```

**Review by Sventimir — CHANGES_REQUESTED** [review]
Looks good except for some minor mistakes that I found.

**Review by joaosreis — CHANGES_REQUESTED** [review]
Seems good, just take a look at the requested changes.

File: docs/exchange-operators/rosetta/build-from-sources.mdx
Comment (barriebyron) [reply docs/exchange-operators/rosetta/build-from-sources.mdx:43]: fwiw @qwadratic auto-numbering does work, even with code snippets 
https://github.com/o1-labs/docs2/wiki/Markdown-Syntax#numbered-steps 

TLDR;  indent snippets

> To generate an ordered list, start each step with 1. so that each step automatically increments numbering. Auto-numbering is respected only when the content under a step is indented (2 or 4 spaces; in IDE such as VS Code, press the tab key).
> 
> The indentation for numbered steps is required to position the content under each step and to keep the auto-numbering in step.

File: docs/exchange-operators/rosetta/index.mdx
Comment (qwadratic) [reply docs/exchange-operators/rosetta/index.mdx:28]: added a sentence that Rosetta is preferred

File: docs/exchange-operators/rosetta/samples/index.mdx
Comment (qwadratic) [reply docs/exchange-operators/rosetta/samples/index.mdx]: removed jq

File: docs/exchange-operators/rosetta/samples/index.mdx
Comment (qwadratic) [reply docs/exchange-operators/rosetta/samples/index.mdx]: changed

**Comment by qwadratic** [comment]
@barriebyron I believe this one is now good to be merged.

**Review by Sventimir — APPROVED** [review]

**Pull request merged**
